fxa-rust-client/src/lib.rs

```diff
@@ -69,13 +84,8 @@ impl FxAWebChannelResponse {
     }
 }
 
-#[derive(Serialize, Deserialize)]
 pub struct FirefoxAccount {
-    config: FxAConfig,
-    uid: String,
-    email: String,
-    state: FxAState,
-    oauth_tokens_cache: HashMap<String, String>,
+    persisted_state: FxAStateV1,
```

Comment:
It's not clear why you're emphasizing _persisted_ here.  Presumably this thing is `mut` some of the time, in which case whatever I'm mutating is definitely not persisted.  Why not just `state`?

And really, don't you want to always have a `FirefoxAccount(FxAStateV1)`, and then do the serialization/deserialization to get whatever you have into that form?  That is, why would I ever want to work with two different `FirefoxAccount` instances with different state internal representations... other than to upgrade one into the other?